### DISCUSSION TITLE
I can't use Plausible in the child theme

### DISCUSSION THREAD

**Discussion by Alrodr** [thuliteio/doks#810] [Q&A] [answered]
Hello!
First of all, thank you for the excellent work you do.

When I want to have Plausible on my website, it asks me the following.

> Add javascript snippet
> Paste this snippet in the <head> of your website.

however it is not possible, since I do not have access to the header section in the child theme.
Any solution?

Thank you!!

**Comment by h-enk** [comment, marked as answer]
This will be easier with Doks v0.5 (available soon) — see also https://github.com/h-enk/doks/pull/755

For now:
- copy `./node_modules/@hyas/doks/layouts/partials/head/script-header.html` to `./layouts/partials/head/script-header.html` and paste in your Plausible snippet
- copy `/node_modules/@hyas/doks/layouts/index.headers` to `/layouts/index.headers` and add `https://plausible.io/api/event` to the `connect-src` part of `Content-Security-Policy`, like so:

  ```bash
  /*
    Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
    X-Content-Type-Options: nosniff
    X-XSS-Protection: 1; mode=block
    Content-Security-Policy: default-src 'self'; frame-ancestors https://jamstackthemes.dev; manifest-src 'self' https://*.netlify.app; connect-src 'self' https://*.netlify.app https://plausible.io/api/event; font-src 'self' https://*.netlify.app; img-src 'self' https://*.netlify.app data:; script-src 'self' https://*.netlify.app 'sha512-RBYr6Ld4w1yVqaACrgrBLQfPgGhj/1jyacA74WxJ1KM6KVcSWymwrdDwb3HDcdpwiNJ5yssot1He0U9vXoQVlg==' 'sha256-aWZ3y/RxbBYKHXH0z8+8ljrHG1mSBvyzSfxSMjBSaXk=' 'sha256-vOgyKS2vkH4n5TxBJpeh9SgzrE6LVGsAeOAvEST6oCc='; style-src 'self' https://*.netlify.app 'unsafe-inline'
    X-Frame-Options: SAMEORIGIN
    Referrer-Policy: strict-origin
    Feature-Policy: geolocation 'self'
    Cache-Control: public, max-age=31536000
    Access-Control-Allow-Origin: *
  ```

**Reply by Alrodr** [reply]
Amazing, I like the solution but I have a problem an another question.
The problem:

> Refused to load the script 'https://plausible.io/js/plausible.js' because it violates the following Content Security Policy directive: "script-src 'self' https://*.netlify.app 'sha512-RBYr6Ld4w1yVqaACrgrBLQfPgGhj/1jyacA74WxJ1KM6KVcSWymwrdDwb3HDcdpwiNJ5yssot1He0U9vXoQVlg==' 'sha256-aWZ3y/RxbBYKHXH0z8+8ljrHG1mSBvyzSfxSMjBSaXk=' 'sha256-vOgyKS2vkH4n5TxBJpeh9SgzrE6LVGsAeOAvEST6oCc='". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.

The question:
If in the future I upgrade to v.0.5, is it a problem to have done this hack? Will I have to remove it?

Thank you very much!!

**Reply by h-enk** [reply]
Add `https://plausible.io/js/plausible.js` to the `script-src` part of `Content-Security-Policy`, like so:

  ```bash
  /*
    Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
    X-Content-Type-Options: nosniff
    X-XSS-Protection: 1; mode=block
    Content-Security-Policy: default-src 'self'; frame-ancestors https://jamstackthemes.dev; manifest-src 'self' https://*.netlify.app; connect-src 'self' https://*.netlify.app https://plausible.io/api/event; font-src 'self' https://*.netlify.app; img-src 'self' https://*.netlify.app data:; script-src 'self' https://*.netlify.app https://plausible.io/js/plausible.js 'sha512-RBYr6Ld4w1yVqaACrgrBLQfPgGhj/1jyacA74WxJ1KM6KVcSWymwrdDwb3HDcdpwiNJ5yssot1He0U9vXoQVlg==' 'sha256-aWZ3y/RxbBYKHXH0z8+8ljrHG1mSBvyzSfxSMjBSaXk=' 'sha256-vOgyKS2vkH4n5TxBJpeh9SgzrE6LVGsAeOAvEST6oCc='; style-src 'self' https://*.netlify.app 'unsafe-inline'
    X-Frame-Options: SAMEORIGIN
    Referrer-Policy: strict-origin
    Feature-Policy: geolocation 'self'
    Cache-Control: public, max-age=31536000
    Access-Control-Allow-Origin: *
  ```

> If in the future I upgrade to v.0.5, is it a problem to have done this hack? Will I have to remove it?

The additions to `index.headers` are how you make your Doks site as secure as possible (now and in the future).

The changes made to `script-header.html` would in future versions of Doks be made to `custom-head.html` in stead. Since I don't expect `script-header.html` to be changed with Doks v0.5, you could leave your changes there (so you don't have to transfer them to `custom-head.html`)

**Reply by h-enk** [reply]
BTW, it's actually not a hack, but how parent-child template overrides work. So, if you'd like to change some (minor) things, you could copy over the corresponding file from `/node_modules/@hyas/doks/` (parent) to `./` (child), and make the (minor) changes there. Good to realize though, that when a parent file is updated, the child equivalent (if present) still prevails (and the parent file changes will not be effective). That's why you'd have to use overrides sparsely + wisely

**Reply by Alrodr** [reply]
Thank you very much for the explanation and the work. I have become a backer to thank you.
Have a nice day
